src/controllers/credentials/CredentialController.ts

```diff
@@ -198,12 +199,17 @@ export class CredentialController extends Controller {
     @Res() internalServerError: TsoaResponse<500, { message: string }>
   ) {
     try {
+      //check connection exists
+      const connection = await this.agent.connections.findById(options.connectionId)
+      if (!connection)
+        return notFoundError(404, { reason: `connection with connection id "${options.connectionId}" not found.` })
```

Comment:
add a test for this `404`?

Reply:
So I did I just don't know how to extract the error message from the response - but idk if we need that. The error response looks like this: ` @Res() notFoundError: TsoaResponse<404, { reason: string }>,`

Reply:
I think you'd look at `response.body` but you'd have to be quite specific to match the `reason`, seems fine as is to me